src/connectors/numeric-menu/connectNumericMenu.js

```diff
@@ -170,80 +170,67 @@ export default function connectNumericMenu(renderFn, unmountFn = noop) {
       },
 
       getWidgetState(uiState, { searchParameters }) {
-        const currentRefinements = searchParameters.getNumericRefinements(
-          attribute
-        );
-        const equal = currentRefinements['='] && currentRefinements['='][0];
+        const values = searchParameters.getNumericRefinements(attribute);
+
+        const equal = values['='] && values['='][0];
+
         if (equal || equal === 0) {
           return {
             ...uiState,
             numericMenu: {
               ...uiState.numericMenu,
-              [attribute]: `${currentRefinements['=']}`,
+              [attribute]: `${values['=']}`,
             },
           };
         }
 
-        const lowerBound =
-          (currentRefinements['>='] && currentRefinements['>='][0]) || '';
-        const upperBound =
-          (currentRefinements['<='] && currentRefinements['<='][0]) || '';
-
-        if (lowerBound !== '' || upperBound !== '') {
-          if (
-            uiState.numericMenu &&
-            uiState.numericMenu[attribute] === `${lowerBound}:${upperBound}`
-          )
-            return uiState;
-          return {
-            ...uiState,
-            numericMenu: {
-              ...uiState.numericMenu,
-              [attribute]: `${lowerBound}:${upperBound}`,
-            },
-          };
+        const min = (values['>='] && values['>='][0]) || '';
+        const max = (values['<='] && values['<='][0]) || '';
+
+        if (min === '' && max === '') {
+          return uiState;
         }
 
-        return uiState;
+        return {
+          ...uiState,
+          numericMenu: {
+            ...uiState.numericMenu,
+            [attribute]: `${min}:${max}`,
+          },
+        };
       },
 
       getWidgetSearchParameters(searchParameters, { uiState }) {
-        let clearedParams = searchParameters.clearRefinements(attribute);
         const value = uiState.numericMenu && uiState.numericMenu[attribute];
 
+        const withoutRefinements = searchParameters.clearRefinements(attribute);
+
         if (!value) {
-          return clearedParams;
+          return withoutRefinements.setQueryParameters({
+            numericRefinements: {
+              ...withoutRefinements.numericRefinements,
+              [attribute]: {},
+            },
+          });
         }
 
-        const valueAsEqual = value.indexOf(':') === -1 && value;
+        const isExact = value.indexOf(':') === -1;
 
-        if (valueAsEqual) {
-          return clearedParams.addNumericRefinement(
-            attribute,
-            '=',
-            valueAsEqual
-          );
+        if (isExact) {
+          return withoutRefinements.addNumericRefinement(attribute, '=', value);
```

Comment:
We should have the same path than for `min`/`max` i.e. `parseFloat` -> `isFiniteNumber `.